fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -314,4 +321,38 @@ public static int readFhirResourceFromHttps(String dataUrl, int numOfLinesToSkip
 
         return parseFailures;
     }
+
+    /**
+     * Validate the input resource and throw if there are validation errors
+     *
+     * @param resource
+     * @throws FHIRValidationException
+     * @throws FHIROperationException
+     */
+    public static List<OperationOutcome.Issue> validateInput(Resource resource)
+            throws FHIRValidationException, FHIROperationException {
+        List<OperationOutcome.Issue> issues = FHIRValidator.validator().validate(resource);
+        if (!issues.isEmpty()) {
+            boolean includesFailure = false;
+            for (OperationOutcome.Issue issue : issues) {
+                if (FHIRUtil.isFailure(issue.getSeverity())) {
+                    includesFailure = true;
+                }
+            }
+
+            if (includesFailure) {
+                throw new FHIROperationException("Input resource failed validation.").withIssue(issues);
+            } else {
+                if (logger.isLoggable(Level.FINE)) {
+                    String info = issues.stream()
+                                .flatMap(issue -> Stream.of(issue.getDetails()))
+                                .flatMap(details -> Stream.of(details.getText()))
+                                .flatMap(text -> Stream.of(text.getValue()))
+                                .collect(Collectors.joining(", "));
+                    logger.fine("Validation warnings for input resource: " + info);
```

Comment:
```suggestion
                    logger.fine("Validation warnings for input resource: [" + info + "]");
```
I suggest adding a delimiter